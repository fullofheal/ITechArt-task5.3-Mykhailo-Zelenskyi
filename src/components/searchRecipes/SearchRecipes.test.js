import { cleanup, screen, render } from "@testing-library/react";
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from "../../store";
import SearchRecipes from "./SearchRecipes";

afterEach(() => {
  cleanup();
});

describe('test if component is rendered on the page', () => {
  test('should render SearchRecipes component', () => {
    render(
    <Provider store={store}>
      <SearchRecipes />
    </Provider>
    );
    const SearchRecipesComponent = screen.getByTestId('SearchRecipes');
    expect(SearchRecipesComponent).toBeInTheDocument();
  });
});

test('RecipesList matches snapshot', () => {
  const tree = renderer.create(
    <Provider store={store}>
        <SearchRecipes />
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});