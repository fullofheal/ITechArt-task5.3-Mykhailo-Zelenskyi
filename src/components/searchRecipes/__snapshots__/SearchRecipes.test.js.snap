// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipesList matches snapshot 1`] = `
<form
  className="search__wrapper"
  data-testid="SearchRecipes"
  onSubmit={[Function]}
>
  <h2>
    Search a new recipe by one of the following criteria:
  </h2>
  <div
    className="search__options"
  >
    <button
      className="search__option"
      onClick={[Function]}
      type="button"
    >
      Category
    </button>
    <button
      className="search__option"
      onClick={[Function]}
      type="button"
    >
      Ingredients
    </button>
    <button
      className="search__option"
      onClick={[Function]}
      type="button"
    >
      Country
    </button>
    <button
      className="search__option"
      onClick={[Function]}
      type="button"
    >
      Random meal
    </button>
  </div>
  <div>
    <div
      className="search__field"
    >
      <div>
        Please specify desired 
        category
        :
      </div>
      <input
        name="Category"
        onChange={[Function]}
        placeholder="e.g. 'beef', 'breakfast', 'side', etc."
        required={true}
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="search__buttons"
  >
    <button
      className="search__submit-btn"
      type="submit"
    >
      Search
    </button>
    <button
      className="search__submit-btn"
      onClick={[Function]}
      type="button"
    >
      Fill out your own recipe
    </button>
  </div>
</form>
`;
